esp/public/media/scripts/program/modules/scheduling_checks.jsx

```diff
@@ -114,3 +160,98 @@ var RefreshButton = React.createClass({
     </button>;
   },
 });
+
+/**
+ * Calls its onClick prop to reset table greying/sorting
+ */
+var ResetButton = React.createClass({
+  propTypes: {
+    onClick: React.PropTypes.func.isRequired,
+  },
+
+  render: function () {
+    return <button onClick={this.props.onClick} className="reset-button">
+      Reset
+    </button>;
+  },
+});
+
+// Modified from react-json-table example code.
+var SelectTable = React.createClass({
+  
+  propTypes: {
+    rows: React.PropTypes.array.isRequired,
+    saveState: React.PropTypes.shape({
+      greyed: React.PropTypes.object.isRequired,
+      sort: React.PropTypes.any.isRequired,
+      reverse: React.PropTypes.bool.isRequired
+    }).isRequired,
+    header: React.PropTypes.bool.isRequired,
+    columns: React.PropTypes.array,
+    updateTableState: React.PropTypes.func.isRequired
+  },
+    
+  getInitialState: function(){
+    // We will store the sorted column and whether each row is greyed out
+    return {sort: this.props.saveState.sort, greyed: this.props.saveState.greyed, reverse: this.props.saveState.reverse};
+  },  
+  render: function(){
+    this.props.updateTableState(this.state);
```

Comment:
I think this might cause problems -- while rendering the SelectTable, it's going to modify the state of the parent, which will force another rerender.  This also really shouldn't be necessary in the first place -- why do you think you need it?